code/modules/mob/living/carbon/superior_animal/xenomorph/ai.dm

```diff
@@ -1,40 +1,38 @@
 /mob/living/carbon/superior_animal/xenomorph/doTargetMessage()
 	. = ..()
 
-	playsound(src.loc, list('sound/xenomorph/alien_growl1.ogg', 'sound/xenomorph/alien_growl2.ogg', 'sound/xenomorph/alien_growl3.ogg', 'sound/xenomorph/alien_hiss1.ogg', 'sound/xenomorph/alien_hiss2.ogg', 'sound/xenomorph/alien_hiss3.ogg'), 120, 1)
+	playsound(loc, list('sound/xenomorph/alien_growl1.ogg', 'sound/xenomorph/alien_growl2.ogg', 'sound/xenomorph/alien_growl3.ogg', 'sound/xenomorph/alien_hiss1.ogg', 'sound/xenomorph/alien_hiss2.ogg', 'sound/xenomorph/alien_hiss3.ogg'), 120, 1)
 
-/mob/living/carbon/superior_animal/xenomorph/death(var/gibbed,var/message = deathmessage)
+/mob/living/carbon/superior_animal/xenomorph/death(gibbed,message = deathmessage)
 	if (stat != DEAD)
 		target_mob = null
 		stance = initial(stance)
 		stop_automated_movement = initial(stop_automated_movement)
 		walk(src, 0)
-		playsound(src.loc, list('sound/xenomorph/alien_death.ogg', 'sound/xenomorph/alien_death2.ogg', 'sound/xenomorph/alien_help1.ogg'), 120, 1)
+		playsound(loc, list('sound/xenomorph/alien_death.ogg', 'sound/xenomorph/alien_death2.ogg', 'sound/xenomorph/alien_help1.ogg'), 120, 1)
 
-		density = 0
+		density = FALSE
 		layer = LYING_MOB_LAYER
 
 	. = ..()
 
-/mob/living/carbon/superior_animal/xenomorph/spitter/boiler/death(var/gibbed,var/message = deathmessage)
+/mob/living/carbon/superior_animal/xenomorph/spitter/boiler/death(gibbed,message = deathmessage)
 	if (stat != DEAD)
 		target_mob = null
 		stance = initial(stance)
 		stop_automated_movement = initial(stop_automated_movement)
 		walk(src, 0)
-		playsound(src.loc, list('sound/xenomorph/alien_death.ogg', 'sound/xenomorph/alien_death2.ogg', 'sound/xenomorph/alien_help1.ogg'))
-		explosion(src.loc, 0,1,3)
+		playsound(loc, list('sound/xenomorph/alien_death.ogg', 'sound/xenomorph/alien_death2.ogg', 'sound/xenomorph/alien_help1.ogg'))
+		explosion(loc, 0,1,3)
 
-		density = 0
+		density = FALSE
 		layer = LYING_MOB_LAYER
 
 	. = ..()
 
-/mob/living/carbon/superior_animal/xenomorph/gib(var/anim = icon_gib, var/do_gibs = 1)
+/mob/living/carbon/superior_animal/xenomorph/gib(anim = icon_gib, do_gibs = 1)
 	if (!anim)
-		anim = 0
-
-	sleep(1)
```

Comment:
without sleep(1) the thing will be dropped instantly btw if you didnt know

Reply:
Aware and intented, I personally think it even looks better this way.

Like if they get gibbed and play the animation you also see the gibs fly around, its neet